src/lib/router.ts

```diff
@@ -85,12 +85,38 @@ type PaginatedMatches = {
   results: Match[];
 };
 
+type FilterProps = { gameId?: number; username?: string };
+
+function getFilter(filterOptions: FilterProps) {
```

Comment:
Maybe this function should be called `getMatchesFilter` and been placed in the `matches.ts` file.